File: training/easy/mime-type.scala
import scala.io.StdIn._
import scala.collection.mutable.HashMap

object Solution extends App {
  val n = readInt
  val q = readInt
  val types: HashMap[String, String] = HashMap()
  for(_ <- 0 until n) {
    val Array(ext, mt) = readLine split " "
    types.put(ext.toUpperCase, mt)
  }
  for(_ <- 0 until q) {
    val fname = readLine
    val idx = fname.lastIndexOf('.')
    println(types.getOrElse(if (idx > -1) { fname.substring(idx + 1).toUpperCase } else { "" }, "UNKNOWN"))
  }
}

File: training/easy/mime-type.kt
import java.util.Scanner

fun main(args : Array<String>) {
    val input = Scanner(System.`in`)
    val n = input.nextInt() // Number of elements which make up the association table.
    val q = input.nextInt() // Number Q of file names to be analyzed.
    val mimeTypes = (0 until n).associate {
        input.next().toLowerCase() to input.next()
    }
    input.nextLine()
    (0 until q).map {
        val fname = input.nextLine() // One file name per line.
        val ext = fname.substring(fname.lastIndexOf('.') + 1).toLowerCase()
        if (ext.length < fname.length) mimeTypes.getOrDefault(ext,"UNKNOWN") else "UNKNOWN"
    } .forEach(::println)
}
